jbpm-flow/src/main/java/org/jbpm/process/instance/impl/FeelReturnValueEvaluator.java

```diff
@@ -0,0 +1,101 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.instance.impl;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.jbpm.process.core.context.variable.VariableScope;
+import org.jbpm.process.instance.context.variable.VariableScopeInstance;
+import org.jbpm.workflow.instance.WorkflowProcessInstance;
+import org.kie.api.runtime.Globals;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.process.ProcessContext;
+import org.kie.dmn.api.core.DMNRuntime;
+import org.kie.dmn.core.impl.DMNRuntimeImpl;
+import org.kie.dmn.feel.FEEL;
+import org.kie.dmn.feel.lang.FEELProfile;
+
+public class FeelReturnValueEvaluator implements ReturnValueEvaluator, Externalizable {
+    
+    private static final long   serialVersionUID = 630l;
+
+    private String expr;
+
+    public FeelReturnValueEvaluator() {
+    }
+
+    public FeelReturnValueEvaluator(String expr) {
+        this.expr = expr;
+    }
+
+    public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+        expr = in.readUTF();
+    }
+
+    public void writeExternal(ObjectOutput out) throws IOException {
+        out.writeUTF( expr );
+    }
+
+    public Object evaluate(ProcessContext context) throws Exception {
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("kcontext", context);  
+        
+        // insert globals into context
+        Globals globals = context.getKieRuntime().getGlobals();
+        
+        if (globals != null && globals.getGlobalKeys() != null) {
+            for (String gKey : globals.getGlobalKeys()) {
+                variables.put(gKey, globals.get(gKey));
+            }
+        }
+        if (context.getProcessInstance() != null && context.getProcessInstance().getProcess() != null) {
+            // insert process variables
+            VariableScopeInstance variableScope = (VariableScopeInstance) ((WorkflowProcessInstance)context.getProcessInstance())
+                    .getContextInstance(VariableScope.VARIABLE_SCOPE);
+    
+            if (variables != null ) {
+                variables.putAll(variableScope.getVariables());
+            }
+        }
+        DMNRuntime runtime = ((KieSession) context.getKieRuntime()).getKieRuntime(DMNRuntime.class);
+        List<FEELProfile> profiles = ((DMNRuntimeImpl) runtime).getProfiles().stream().map(dmnProfile -> (FEELProfile) dmnProfile).collect(Collectors.toList());
+        FEEL feel = FEEL.newInstance(runtime.getRootClassLoader(), profiles);
```

Comment:
@etirelli @tarilabs would it be possible to expose getProfiles method via DMNRuntime interface? I think this would be the best way to use it and I'd prefer to not use Impl class here...wdyt?

Reply:
I'lll need @etirelli approval to make DMNProfile and FEELProfile in public API too :)

Just a potential suggestion for the Stream, if you want you can replace the lambda with a `FEELProfile.class::cast` or potentially just faster globally given this is one of the limits of generics in Java, the whole line as:
```
List<FEELProfile> profiles = (List)((DMNRuntimeImpl) runtime).getProfiles();
```
I know is ugly, but that's the way with Java is :)

Reply:
@tarilabs thanks, I went the ugly way ... planed to do that from the beginning so it's good to see I am not the only one ;)

let's way for @etirelli opinion about the getProfiles method.

Reply:
@tarilabs you are the lead engineer, so it is your call! ;)

Personally, I think the interfaces should be made public, as we will eventually have other cases where it will be necessary, and casting down to the implementation is never good.